new/language/patternquery/builder/builder.go

```diff
@@ -107,13 +109,53 @@ func (builder *builder) build(rootNode *tree.Node) (*Result, error) {
 	paramToVariable, equalParams := builder.processVariableToParams()
 
 	return &Result{
-		Query:           builder.stringBuilder.String(),
-		ParamToVariable: paramToVariable,
-		EqualParams:     equalParams,
-		ParamToContent:  builder.paramToContent,
+		Query:             builder.stringBuilder.String(),
+		ParamToVariable:   paramToVariable,
+		EqualParams:       equalParams,
+		ParamToContent:    builder.paramToContent,
+		SkipRulesForNodes: builder.skipRulesForNodes(rootNode),
 	}, nil
 }
 
+func (builder *builder) skipRulesForNodes(rootNode *tree.Node) map[string][]*tree.Node {
```

Comment:
This logic should be applied to the source code and not the patterns. Rather than pattern query builder, you want to do it here:

https://github.com/Bearer/bearer/blob/675ed2b41c97c701d75d689771309370ad8f97e2/new/detector/composition/javascript/javascript.go#L183

(and similar for Ruby)